lib/home_page/home.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+
+import '../constants/constants.dart' as constants;
+import '../constants/strings.dart' as strings;
+import '../saved_page/saved_page.dart';
+import '../search_page/search_page.dart';
+import 'exit_alert.dart';
+import 'view_page.dart';
+
+class Home extends StatefulWidget {
+  const Home({super.key});
+
+  @override
+  State<Home> createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  static const int initialIndex = 0;
+  late int _selectedIndex;
+  late List<Widget> _pages;
+  late PageController _pageController;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedIndex = initialIndex;
+    _pages = [
+      const ViewPage(),
+      const SearchPage(),
+      const SavedPage(),
+    ];
+    _pageController = PageController(initialPage: _selectedIndex);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _pageController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () async {
+        if (_selectedIndex != initialIndex) {
+          setState(() {
+            _selectedIndex = initialIndex;
+            _pageController.jumpToPage(_selectedIndex);
+          });
+          return false;
+        } else {
+          final shouldPop = await showDialog<bool>(
+            context: context,
+            builder: (context) {
+              return const ExitAlert();
+            },
+          );
+          return shouldPop ?? false;
+        }
```

Comment:
Small improvement: you could extract this as a separate method, to improve the readability of the `build` method.